### DISCUSSION TITLE
Nextflow failed to deploy a container

### DISCUSSION THREAD

**Discussion by chenfengMeng** [nextflow-io/nextflow#3717] [Q&A]
## Bug report 

When I trying to deploy a container called "quay.io/wtsicgp/cgp-telomerehunter:1.1.0", it failed to deploy it on google cloud even if I don't run any command. but when I run it locally, it work successfully. 

### Expected behavior and actual behavior
It should return some success result. 

### Steps to reproduce the problem

process Telomerehunter {
container  "quay.io/wtsicgp/cgp-telomerehunter:1.1.0"
disk 50 gb

script
"""
"""
}

### Program output 

Error executing process > 'Telomerehunter'

Caused by:
  Process `Telomerehunter` terminated with an error exit status (9)

Command executed:

Command exit status:
  9

Command output:
  (empty)

Command error:
  Execution failed: generic::failed_precondition: while running "nf-5665840d56f2d3211a53347091a50bf2-main": unexpected exit status 1 was not ignored

Work dir:
  gs://repare-nextflow-work/work_dir/56/65840d56f2d3211a53347091a50bf2

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

**Comment by mribeirodantas** [comment]
Hello, @chenfengMeng!

1. How you're using Google Cloud? Google Life Sciences? Google Cloud Batch?
2. Can you share a minimal reproducible example?

**Comment by chenfengMeng** [comment]
Thanks for your fast reply, We are using google life sciences. this is the setting of my GCP,       
     process.executor = "google-lifesciences"
       workDir          = "gs://someplace/work_dir"
       google.location  = "us-central1"
       google.region    = "us-central1"
       google.project   = "your-project"

**Comment by chenfengMeng** [comment]
the minimum reproducible example:
**### show.nf:**
```
nextflow.enable.dsl=1
params.outdir                             = false        //the path to the output directory


process Telomerehunter {
        // primarily take from nf-core sarez

        container "quay.io/wtsicgp/cgp-telomerehunter:1.1.0"
        containerOptions "--user root"

        disk 50.GB

        script:
        """
        touch *.bai
        touch *.bam
        """
}
```

**Reply by chenfengMeng** [reply]
##show.config

```
docker{
    enabled = true
}
profiles {
  gcp {
       	process.executor = "google-lifesciences"
       	workDir          = "gs://someplace/work_dir"
       	google.project   = "your project"
	google.region    = "us-central1"
  }
  localRef {
        path_to_refs              = "." //"[/path/to/refs]"
        params.ref_genome         = "${path_to_refs}/dnaseq/hg19/invitae/hg19.fa"
        params.ref_genome_index   = "${path_to_refs}/dnaseq/hg19/invitae/hg19.fa.fai"
        params.ref_genome_dict    = "${path_to_refs}/dnaseq/hg19/invitae/hg19.fa.dict"
  }
}


process {
	cpu = 1
	memory = 5.GB
}
```

Notice you may want to customize the project and work directory under your preferred.

**Reply by mribeirodantas** [reply]
Is there any reason you must use GLS instead of Google Batch? A lot of issues have been solved by using the latter

**Comment by mribeirodantas** [comment]
This seems to be a problem with the container image. See this issue: https://github.com/nf-core/rnaseq/issues/764

**Comment by chenfengMeng** [comment]
Thanks for your reply and instruction! and May I know what kind of image has this problem? I mean, this image works just fine in my local computer, but, when I deploy it in google cloud, it broken as a result.

**Reply by mribeirodantas** [reply]
I'm sorry but I'm not aware of what issue this container image has. I just found the discussion where the container image was identified as the issue.

**Reply by ptn24** [reply]
@chenfengMeng : Did you ever figure out what the problem was? I think I have the same problem.

**Comment by chenfengMeng** [comment]
I just created a new docker by myself, it should be the problem of
administrator accessibility in the docker, check the layer of the docker
construction.

**Reply by ptn24** [reply]
I see. Can you elaborate on what lines you added to get the image to work? I also have a custom image, but I am not sure where to begin.

**Comment by chenfengMeng** [comment]
I don't know, I just rebuilt one and then it worked.

**Reply by ptn24** [reply]
I see. Thank you!
